hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -165,6 +165,10 @@ private void initIfNeeded() {
           logRecords.put(entry.getKey(), entry.getValue());
         }
       }
+    } else {
```

Comment:
@nsivabalan @codope why are these changes needed again?

Reply:
a test was failing in CI and so I fixed it in this patch. Did not want to land this patch with failing CI. I thought of putting up a separate patch, but the other one will depend on this patch since tests might time out.